### ISSUE TITLE
RTSPtoWeb - WebRTC Addon Stopped Working After Upgrade to 2022.3 - Starts, then fails to add integration in HA UI

### ISSUE THREAD

**Issue by ziptbm** [allenporter/stream-addons#61]
After upgrading to 2022.3, the RTSPtoWeb - WebRTC (version 1.2.1) stopped working and shows 0% resource use.  As part of the troubleshooting, I uninstalled the addon, then reinstalled it.  Upon reinstall and start, I get the following in the logs for the addon and the resources spike for a moment and return to 0%.  I'm running native HA OS on an Intel NUC.

````
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-rtsp-to-web: executing... 
[22:05:37] INFO: Updating configuration
[cont-init.d] 01-rtsp-to-web: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[22:05:37] INFO: Starting discovery service
[22:05:37] INFO: Starting RSTPtoWeb
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[22:05:37] INFO: Successfully send discovery information to Home Assistant (8083).
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
````

The Integrations screen in the UI of HA discovers the service, but it fails to load upon adding.  And also navigating to the web UI for the addon also fails to load.  As part of troubleshooting, I've also tried enabling and disabling watchdog, but it appears it needs to be off per some of the various threads I've found on this topic.

Any ideas what is going on and why this addon is no longer working?

**Comment by allenporter** [comment]
So you have information about the integration failing to load?

How long is it before those last two lines are logged after the add-on is started?

How about turning up debug mode. That may give us a clue for what the add-on is up to.

Thanks!

**Comment by ziptbm** [comment]
Thank you.  Please let me know if I’ve missed something. 


> How long is it before those last two lines are logged after the add-on is started?

Within a couple of seconds.


I turned on debug, stopped/started the add on and for the following in the logs. 

```
`[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-rtsp-to-web: executing... 
[23:07:08] INFO: Updating configuration
[cont-init.d] 01-rtsp-to-web: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[23:07:08] INFO: Starting discovery service
[23:07:08] INFO: Starting RSTPtoWeb
time="2022-03-27T23:07:08-04:00" level=info msg="Server CORE start" func=main module=main
time="2022-03-27T23:07:08-04:00" level=info msg="Server RTSP start" call=Start func=RTSPServer module=rtsp_server
time="2022-03-27T23:07:08-04:00" level=info msg="Server start success a wait signals" func=main module=main
time="2022-03-27T23:07:08-04:00" level=info msg="Server HTTP start" call=Start func=RTSPServer module=http_server
time="2022-03-27T23:07:08-04:00" level=fatal msg="listen tcp :8083: bind: address already in use" call=ServerHTTPPort func=HTTPAPIServer module=http_router
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[23:07:08] INFO: Successfully send discovery information to Home Assistant (8083).
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting. `
```

**Comment by allenporter** [comment]
Are you running another instance of this service? The port is in use.

**Comment by ziptbm** [comment]
> Are you running another instance of this service? The port is in use.

Not that I know of...

I wonder if my troubleshooting attempt for uninstalling it didn't work or something.  Continuing to look as I don't understand why it would think there's another instance of it running.

**Comment by allenporter** [comment]
You could try adjusting the port in the mean time.

**Comment by MRobi1** [comment]
> You could try adjusting the port in the mean time.

Just ran into this same issue and moved the port to 8084 and it worked first shot.
I did check my other addons and none show port 8083, so not sure what's using it but changing the port does indeed work.

**Comment by MRobi1** [comment]
@allenporter one note that I did notice, after changing the port to 8084, if I click "Open WEB UI" it still attempts to open a new window directing it to port 8083. I have to manually change to 8084 in the address bar.

**Comment by allenporter** [comment]
Yes, you'll have to manually fix the port yourself in the url bar due to limitations in how add-ons use host network and port configuration. We can't update the url if you change the port.

I suggest you figure out what is open on port 8083 if you want to make that work.

Glad your issue was resolved.

**Comment by MRobi1** [comment]
So I ran into this issue again. This time while on port 8084. Changing it to 8085 made it work again.

I've added nothing to my installation since changing to port 8084. There are no addons configured on either 8083 or 8084. All I did was restart in between.

So I did some further testing. Got it running with 8085 and restarted the hardware (the physical machine and not just restarting HA). On reboot, same error on port 8085. So now ports 8083-8085 are causing the same issue. Set to 8086 and it starts again. Rebooted hardware, same error coming up on 8086.

So now I'm running again on 8087, but it looks like on every hardware reboot I need to change the port.

**Comment by allenporter** [comment]
Silly question but can you go back to 8083?

**Comment by MRobi1** [comment]
Nope, can't go back to any of the previously used ports.

Now what I haven't tried is completely uninstalling the add-on, restarting and re-installing it to see if it opens them back up.

**Comment by kongo09** [comment]
I seem to have the same problem. Changing over to 8084 made it start. Running latest HA on a RPi4.

**Comment by allenporter** [comment]
I'm unable to reproduce this problem. Something outside of the add-on is using the ports, so i'm not sure what the add-on can do about this. It only uses one port you specify.

**Comment by MRobi1** [comment]
I can say with complete confidence that nothing else is using port 8083 on my installation.
 
If there were a port conflict, it would not have worked initially on 8083. Further to that, after changing it to port 8084 it worked again. However another reboot and 8084 was suddenly "in use". Same with 8085 and 8086. These ports aren't going to be suddenly and coincidentally taken by something else when the only thing happening is a reboot. 

Ultimately, I worked my way up to 8087 fighting this issue and it's been good for a few months now.

**Comment by allenporter** [comment]
Right, i know what you're saying -- the error message still `listen tcp :808X: bind: address already in use`? Clearly *something* is using it, just we can't find it. The add-on only knows how to listen on a single port as far as I know.

What is needed is to actually get into the host OS or docker container and run the unix commands to see which services are listening on which ports.  Perhaps the SSH add-on can help? then something debugging like these cases e.g. https://community.home-assistant.io/t/hassio-mosquito-port-1883-in-use/68537/7

**Comment by allenporter** [comment]
I'm curious to learn from folks what kinds of hardware they are using when seeing this issue? I have still yet to reproduce it.

**Comment by fribse** [comment]
Proxmox server 7.2-7
HA VM: 2 cores, 10 GB Ram, 150 GB Disk
Home Assistant 2022.10.2 Supervisor 2022.10.0 Operating System 9.0
RTSPtoWeb: 1.4.0

After updating to 1.4.0 and the old problem was solved, I see this one as well.

```
[08:52:24] INFO: Starting RSTPtoWeb
time="2022-10-11T08:52:24+02:00" level=info msg="Server CORE start" func=main module=main
time="2022-10-11T08:52:24+02:00" level=info msg="Server RTSP start" call=Start func=RTSPServer module=rtsp_server
time="2022-10-11T08:52:24+02:00" level=info msg="Server start success a wait signals" func=main module=main
time="2022-10-11T08:52:24+02:00" level=info msg="Server HTTP start" call=Start func=RTSPServer module=http_server
time="2022-10-11T08:52:24+02:00" level=fatal msg="listen tcp :8083: bind: address already in use" call=ServerHTTPPort func=HTTPAPIServer module=http_router
[06:52:24] WARNING: Halt add-on with exit code 1
```
I tried using
`netstat -tulpn | grep LISTEN`
from the SSH addon, but didn't see anything on that port.

```
[core-ssh ~]$ netstat -tulpn | grep LISTEN
tcp        0      0 127.0.0.11:46469        0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:8099            0.0.0.0:*               LISTEN      139/ttyd
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      138/sshd -D -e [lis
tcp        0      0 :::22                   :::*                    LISTEN      138/sshd -D -e [lis

```

**Comment by atv2016** [comment]
I get the port issue as well, can't start the add-on because something is using 8083.

**Comment by atv2016** [comment]
When i connect to the ip address in question it says: Server unexpectedly dropped connection.

This is what i see in my ssh add-on output from netstat -a, something is definitely bound to 127.0.0.1:8083:
unix  3      [ ]         STREAM     CONNECTED     3854182  /var/run/docker/containerd/containerd.sock.ttrpc
unix  3      [ ]         STREAM     CONNECTED     14310    /run/containerd/s/d4fdf886f27858a8c18637cdec7a29dc78d4cee0afe22350224ade2d48139d15
unix  3      [ ]         STREAM     CONNECTED     11731    
unix  3      [ ]         STREAM     CONNECTED     10154    
unix  3      [ ]         STREAM     CONNECTED     11087    
unix  3      [ ]         STREAM     CONNECTED     14969    /var/run/docker/containerd/containerd.sock.ttrpc
unix  3      [ ]         STREAM     CONNECTED     9822     /run/systemd/journal/stdout
unix  2      [ ]         DGRAM      CONNECTED     11154    
unix  3      [ ]         DGRAM      CONNECTED     11160    
unix  3      [ ]         STREAM     CONNECTED     8337     /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     10180    
unix  3      [ ]         STREAM     CONNECTED     9971     /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     11050    
unix  3      [ ]         STREAM     CONNECTED     24436    
unix  3      [ ]         STREAM     CONNECTED     23531    
unix  3      [ ]         STREAM     CONNECTED     3494578  
unix  3      [ ]         STREAM     CONNECTED     13994    
unix  3      [ ]         STREAM     CONNECTED     933      
unix  2      [ ]         DGRAM      CONNECTED     8296     
unix  3      [ ]         STREAM     CONNECTED     8931     
unix  3      [ ]         STREAM     CONNECTED     8376     
unix  2      [ ]         DGRAM      CONNECTED     11077    
unix  2      [ ]         DGRAM                    9860     
unix  3      [ ]         STREAM     CONNECTED     26571    
unix  3      [ ]         STREAM     CONNECTED     26559    
unix  3      [ ]         STREAM     CONNECTED     3498131  
unix  3      [ ]         STREAM     CONNECTED     3854033  /run/containerd/s/46c707b11c23745e6365ea1bfda61ff7b6e7503c7338bb3e7f38e1d196ed4ac8
unix  3      [ ]         STREAM     CONNECTED     8325     
unix  3      [ ]         STREAM     CONNECTED     11664    
unix  2      [ ]         DGRAM      CONNECTED     8397     
unix  3      [ ]         STREAM     CONNECTED     11134    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     3498139  
unix  3      [ ]         STREAM     CONNECTED     13779    
unix  3      [ ]         STREAM     CONNECTED     3495509  
unix  3      [ ]         STREAM     CONNECTED     2633388  
unix  3      [ ]         STREAM     CONNECTED     11237    
unix  3      [ ]         STREAM     CONNECTED     11876    
unix  3      [ ]         DGRAM      CONNECTED     8242     
unix  3      [ ]         STREAM     CONNECTED     12088    
unix  3      [ ]         STREAM     CONNECTED     8909     
unix  3      [ ]         STREAM     CONNECTED     9811     
➜  ~ netstat -a | grep 8083
tcp        0      0 localhost:8083          0.0.0.0:*               LISTEN     
tcp        0      0 localhost:8083          localhost:53638         ESTABLISHED
tcp        0      0 localhost:8083          localhost:53632         ESTABLISHED
tcp    67666      0 localhost:53632         localhost:8083          ESTABLISHED
tcp    67666      0 localhost:53638         localhost:8083          ESTABLISHED  
                                                                                   
Try `wget --help' for more options.
➜  ~ wget localhost:8083
--2022-11-09 09:20:54--  http://localhost:8083/
Resolving localhost (localhost)... ::1, 127.0.0.1
Connecting to localhost (localhost)|::1|:8083... failed: Connection refused.
Connecting to localhost (localhost)|127.0.0.1|:8083... connected.
HTTP request sent, awaiting response... 404 Not Found
2022-11-09 09:20:54 ERROR 404: Not Found.

**Comment by allenporter** [comment]
As I review issue with docker, it seems like its possible something that docker has had a bug with in the past. I am fairly positive this is not an issue with this container (e.g. it shouldn't be possible for a docker container to make a port in use after a restart). I think someone should try investigating with a docker bug report as a next step.
